onnxruntime/core/session/inference_session.cc

```diff
@@ -422,6 +422,24 @@ common::Status InferenceSession::Initialize() {
                                                    std::make_unique<CPUExecutionProvider>(epi)));
     }
 
+    if (!session_options_.enable_sequential_execution) {
+      for (auto it = execution_providers_.begin(); it != execution_providers_.end(); ++it) {
+        // very rare case - user has registered CPU provider first -
+        // so parallel execution is still possible as all nodes will get assigned CPU provider
+        if (it->get()->Type() == onnxruntime::kCpuExecutionProvider) {
+          break;
+        }
+
+        if (it->get()->Type() == onnxruntime::kCudaExecutionProvider) {
```

Comment:
@jywu-msft do other execution providers work with the parallel executor? i assume not. This line is checking for cuda only today.

Reply:
cuda explicitly can't support parallel executor due to implementation using only one compute stream.
for mkldnn, ngraph etc. you can run parallel executor. we just need to make sure there isn't an oversubscription of threads by limiting number of OMP threads they can spawn.